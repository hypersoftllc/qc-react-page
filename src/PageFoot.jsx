import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function PageFoot(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`.trim(),
  }
  delete domProps.children
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

PageFoot.defaultProps = {
  className: '',
  compClassName: 'Page-Foot',
  compType: 'footer',
}

PageFoot.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf(['div', 'footer', 'section']),
}

export default PageFoot
