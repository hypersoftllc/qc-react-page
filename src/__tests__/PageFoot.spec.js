/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import PageFoot from '../PageFoot'


describe('PageFoot', () => {

  it('should be a function', () => {
    expect(typeof PageFoot).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <PageFoot/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageFoot/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageFoot>
            <span>Child</span>
          </PageFoot>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageFoot className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageFoot compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageFoot className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageFoot className="foo" compClassName="page-foot"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compType', () => {

      it('should render with expected component', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageFoot compType="div"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageFoot compType="footer"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageFoot compType="section"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
