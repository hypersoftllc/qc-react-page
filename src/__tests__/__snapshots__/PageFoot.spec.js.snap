// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFoot rendered with children should render children 1`] = `
<footer
  className="Block Page-Foot"
>
  <span>
    Child
  </span>
</footer>
`;

exports[`PageFoot rendered with className should render with expected class attribute 1`] = `
<footer
  className="Block Page-Foot Foo"
/>
`;

exports[`PageFoot rendered with compClassName should render with expected class attribute 1`] = `
<footer
  className="Block"
/>
`;

exports[`PageFoot rendered with compClassName should render with expected class attribute 2`] = `
<footer
  className="Block  foo"
/>
`;

exports[`PageFoot rendered with compClassName should render with expected class attribute 3`] = `
<footer
  className="Block page-foot foo"
/>
`;

exports[`PageFoot rendered with compType should render with expected component 1`] = `
<div
  className="Block Page-Foot"
/>
`;

exports[`PageFoot rendered with compType should render with expected component 2`] = `
<footer
  className="Block Page-Foot"
/>
`;

exports[`PageFoot rendered with compType should render with expected component 3`] = `
<section
  className="Block Page-Foot"
/>
`;

exports[`PageFoot rendered with no props should use default props 1`] = `
<footer
  className="Block Page-Foot"
/>
`;
